ui/app/rif/pages/addNewMulticryptoAddress/addNewMulticryptoAddress.js

```diff
@@ -0,0 +1,137 @@
+			if(selectFirst){
+				selected = key
+				selectFirst = false
+			}
+		});
+		this.state = { 
+			networks: [...networks],
+			network: selected,
+		};
+	}
+
+	navigateTo (url) {
+		global.platform.openWindow({ url })
+	}
+	addAddress = () => {
+		let domains = JSON.parse(localStorage.rnsDomains);
+		let myIndex = -1
+		let resolverIndex = this.props.selectedResolver
+		let selecteddomain = domains.find((domain, index) => {
+			if(domain.domain === this.props.domain.domain){
+				myIndex = index
+				return domain
+			}
+		})
+		if(myIndex !== -1 && resolverIndex !== -1){
+			let newNetwork = {
+				networkName: this.state.network,
+				networkIcon:this.state.network,
+				address: this.state.address,
+			}
+			domains[myIndex].resolvers[resolverIndex].network.push(newNetwork)
+			localStorage.setItem('rnsDomains', JSON.stringify(domains))
+			//Sending back with localstorage rnsDomains (Here we try to get again localstorage so if it wasnt updated, we're going to show whats really saved)
+			domains = JSON.parse(localStorage.rnsDomains)
+			this.props.goBack(domains[myIndex])
+		}
+	}
+	_updateAddress = (e) => {
+		this.setState({address: e.target.value})
+	}
+	_updateNetwork = (e) => {
+		console.log("e.target.value", e.target.value)
+		this.setState({network: e.target.value})
+	}
+	render () {
+		return (
+		<div className={'body'}>
+			<FontAwesomeIcon icon={faChevronLeft} className={'rif-back-button'} onClick={() => this.props.goBack(this.props.domain)}/>
+			<div style={{position: 'relative',}}>
+				<div id='title' className={'full-width add-new-multicrypto-title'}>
+					<span>Add new network</span>
+				</div>
+				<div id='comboNetworks' className={'add-new-multicrypto-select'}>
+					<select id='comboNetworks' className="select-css" onChange={this._updateNetwork}>
+						{this.state.networks.map((network, index) => {
+								return <option key={index} value={network}>{network.toUpperCase()}</option>
+							})
+						}	
+					</select>
+				</div>
+				<div id='inputAddress' className={'full-width add-new-multicrypto-input'}>
+					<input type='text' placeholder="value" onChange={this._updateAddress}/>
+				</div>
+				<div id='bottomForm' className={'full-width add-new-multicrypto-bottom'}>
+					<div id='buttonCancel' className={'add-new-multicrypto-button-cancel'}>
+						<CustomButton 
+							text={'CANCEL'}
+							onClick={() => this.props.goBack(this.props.domain)} 
+							className={
+								{
+									button: 'custom-button-cancel center',
+									text: 'center',
+								}
+							}
+						/>
+					</div>
+					<div id='buttonSave' className={'add-new-multicrypto-button-save'}>
+						<CustomButton 
+							text={'SAVE'}
+							onClick={() => this.addAddress()} 
+							className={
+								{
+									button: 'custom-button-success center',
+									text: 'center',
+								}
+							}
+						/>
+					</div>
+				</div>
+			</div>
+		</div>
+		)
+	}
+}
+
+AddNewMulticryptoAddressScreen.propTypes = {
```

Comment:
You should put this inside the component class

Reply:
Yep, it was re-coded to be more like the new convention as mentioned in other comment